innopoints/core/notifications.py

```diff
@@ -0,0 +1,34 @@
+"""Helper module for sending notifications, depending on the user preference"""
+
+import logging
+from typing import List
+
+from sqlalchemy.exc import IntegrityError
+
+from innopoints.extensions import db
+from innopoints.models import Notification
+
+log = logging.getLogger(__name__)
+
+
+def notify(recipient_email: str, notification_type: str, payload=None):
+    """Sends a notification to the specified user."""
+    # TODO: check preferences
+    # TODO: send Email
+    # TODO: send Push
+    notification = Notification(
+        recipient_email=recipient_email,
+        type=notification_type,
+        payload=payload,
+    )
+    try:
+        db.session.add(notification)
+        db.session.commit()
+        log.info(f'Sent a notification to {recipient_email}')
+    except IntegrityError as exc:
+        log.exception(exc)
+
+def notify_all(recipient_emails: List[str], notification_type: str, payload=None):
```

Comment:
This function looks awfully simple, especially considering that you end up looping twice: once to get emails and once to send notifications. Maybe have it accept a list of users instead?